.github/workflows/compileLint.yml

```
@@ -1,18 +1,24 @@
-name: Lint
+name: Compile and Lint
 on: [push, pull_request]
 jobs:
-  lint:
-    name: Lint
+  compileLint:
+    name: Compile and Lint
     runs-on: ubuntu-latest
     steps:
       - name: Checkout Project
         uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
       - name: Use Node.js 16
         uses: actions/setup-node@v3.3.0
         with:
           node-version: 16
           cache: "yarn"
       - name: Install dependencies
         run: yarn
-      - name: Lint
+      - name: Compile changed files
+        run: |
+          git config --global core.quotePath false
+          yarn compile ${{ github.event_name }} $GITHUB_BASE_REF
+      - name: Check Lint
```

Original sounds weird as you're liniting the code
```suggestion
      - name: Linter
```

---

You’re checking if the code was linted correctly, you’re not linting it

---

Isn't a linter a lint checker anyway?